File: test_harness/test_suites/typescript_order_processing_v001/repository.ts
import { Order } from './models';

export interface IOrderRepository {
  save(order: Order): string;
  findById(orderId: string): Order | undefined;
  findByCustomer(customerId: string): Order[];
}

export class OrderRepository implements IOrderRepository {
  private store: Map<string, Order> = new Map();

  save(order: Order): string {
    this.store.set(order.orderId, order);
    return order.orderId;
  }

  findById(orderId: string): Order | undefined {
    return this.store.get(orderId);
  }

  findByCustomer(customerId: string): Order[] {
    return Array.from(this.store.values()).filter(
      (o) => o.customerId === customerId,
    );
  }
}

File: test_harness/test_suites/python_order_processing_v001/repository.py
from abc import ABC, abstractmethod
from typing import List, Optional
from models import Order


class IOrderRepository(ABC):
    @abstractmethod
    def save(self, order: Order) -> str:
        ...

    @abstractmethod
    def find_by_id(self, order_id: str) -> Optional[Order]:
        ...

    @abstractmethod
    def find_by_customer(self, customer_id: str) -> List[Order]:
        ...


class OrderRepository(IOrderRepository):
    def __init__(self) -> None:
        self._store: dict[str, Order] = {}

    def save(self, order: Order) -> str:
        self._store[order.order_id] = order
        return order.order_id

    def find_by_id(self, order_id: str) -> Optional[Order]:
        return self._store.get(order_id)

    def find_by_customer(self, customer_id: str) -> List[Order]:
        return [o for o in self._store.values() if o.customer_id == customer_id]
